src/main/java/ca/on/oicr/pineryreports/Main.java

```diff
@@ -0,0 +1,146 @@
+package ca.on.oicr.pineryreports;
+
+import java.io.File;
+import java.io.IOException;
+
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.CommandLineParser;
+import org.apache.commons.cli.DefaultParser;
+import org.apache.commons.cli.HelpFormatter;
+import org.apache.commons.cli.Option;
+import org.apache.commons.cli.Options;
+import org.apache.commons.cli.ParseException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ca.on.oicr.pinery.client.HttpResponseException;
+import ca.on.oicr.pinery.client.PineryClient;
+import ca.on.oicr.pineryreports.data.ReportFormat;
+import ca.on.oicr.pineryreports.reports.Report;
+import ca.on.oicr.pineryreports.reports.impl.GeccoReport;
+import ca.on.oicr.pineryreports.reports.impl.StockReport;
+
+
+public class Main {
+  
+  private static final Logger LOG = LoggerFactory.getLogger(Main.class);
+  
+  private static final String OPT_SOURCE = "source";
+  private static final String OPT_REPORT = "report";
+  private static final String OPT_FORMAT = "format";
+  private static final String OPT_OUTFILE = "outfile";
+  
+  public static void main(String[] args) {
+    Options opts = getMainOptions();
+    Report report = null;
+    try {
+      CommandLine mainCommand = getCommandLine(args, opts, false);
+      
+      String pineryOpt = mainCommand.getOptionValue(OPT_SOURCE);
+      try (PineryClient pinery = new PineryClient(pineryOpt)) {
+        String reportOpt = mainCommand.getOptionValue(OPT_REPORT);
+        report = getReport(reportOpt);
+        
+        String formatOpt = mainCommand.getOptionValue(OPT_FORMAT);
+        ReportFormat format = formatOpt == null ? report.getDefaultFormat() : ReportFormat.get(formatOpt);
+        if (!report.getValidFormats().contains(format)) {
+          throw new ParseException("Invalid format for this report: " + formatOpt);
+        }
+        
+        String outputOpt = mainCommand.getOptionValue(OPT_OUTFILE, "report" + format.getExtension());
+        File outFile = new File(outputOpt);
+        if (outFile.exists()) {
+          throw new ParseException("Output file already exists: " + outputOpt);
+        }
+        
+        for (Option opt : report.getOptions()) {
+          opts.addOption(opt);
+        }
+        CommandLine reportCommand = getCommandLine(args, opts, true);
+        report.processOptions(reportCommand);
+        
+        LOG.info("Options ok. Generating " + report.getTitle() + "...");
+        report.generate(pinery, format, outFile);
+        LOG.info("Report generated: " + outFile.getName());
+      }
+    } catch (ParseException e) {
+      LOG.error(e.getMessage());
+      showHelp(report);
+      System.exit(1);
+    } catch (HttpResponseException e) {
+      LOG.error("Error fetching data from Pinery");
+      e.printStackTrace();
+      System.exit(2);
+    } catch (IOException e) {
+      LOG.error("Error writing to file");
+      e.printStackTrace();
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Use a logger to log this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1148)